airflow/contrib/sensors/weekday_sensor.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import calendar
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils import timezone
+from airflow.utils.decorators import apply_defaults
+
+
+class DayOfWeekSensor(BaseSensorOperator):
+    """
+    Waits until the first specified day of the week. For example, if the execution
+    day of the task is '2018-12-22' (Saturday) and you pass '5', the task will wait
+    until next Friday (Week Day: 5)
+
+    :param week_day_number: Day of the week as an integer, where Monday is 1 and
+        Sunday is 7 (ISO Week Numbering)
+    :type week_day_number: int
+    :param use_task_execution_day: If ``True``, uses task's execution day to compare
+        with week_day_number. Execution Date is Useful for backfilling.
+        If ``False``, uses system's day of the week. Useful when you
+        don't want to run anything on weekdays on the system.
+    :type use_task_execution_day: bool
+    """
+
+    @apply_defaults
+    def __init__(self, week_day_number,
+                 use_task_execution_day=False,
+                 *args, **kwargs):
+        super(DayOfWeekSensor, self).__init__(*args, **kwargs)
+        self.week_day_number = week_day_number
+        self.use_task_execution_day = use_task_execution_day
+
+    def poke(self, context):
+        if self.week_day_number > 7:
+            raise ValueError(
+                'Invalid value ({}) for week_day_number! '
+                'Valid value: 1 <= week_day_number <= 7'.format(self.week_day_number))
+        self.log.info('Poking until weekday is %s, Today is %s',
+                      calendar.day_name[self.week_day_number - 1],
+                      calendar.day_name[timezone.utcnow().isoweekday() - 1])
+        if self.use_task_execution_day:
+            return context['execution_date'].isoweekday() == self.week_day_number
+        else:
+            return timezone.utcnow().isoweekday() == self.week_day_number
+
+
+class WeekEndSensor(BaseSensorOperator):
```

Comment:
Please move this class to a separate file.

Reply:
My reasoning behind keeping them in the same file is because they are related. But I can move it into a separate file if you think that would be better.

Reply:
My experience with multiple classes in a single file it that it will grow quickly, and we end up with a lot of sensors in a single file.

Reply:
Lets hear what @ashb his opinion is :-)

Reply:
Agree to @Fokko. In my opinion multiple classes can be in a single file when there is only one of them public. And so both sensors are public / do not directly depend on each other they should be in separate files.

@kaxil I think **relations** should be recognizable by **packages** not by files.

Reply:
Well, I agree partially to it but Python gives you liberty to group classes into a module which is also a notion for Django where related classes are inside a module, which makes it easy to maintain. If the files is huge like our `models.py`, I definitely agree that it should be split into multiple files.

Some links where I will try to convince you guys 😀:
- https://stackoverflow.com/a/107836/5691525

-https://stackoverflow.com/questions/1091756/are-multiple-classes-in-a-single-file-recommended (Has some good answers)

There would tons of files if we stick to one class per module notion and it would make "imports" too noisy.

I am happy to move it to separate classes if you guys are not convinced 😉

Reply:
IMO, in Python the general preference is to group classes/functions with related (coherent!) functionality in a single module. For example, scikit-learn also groups multiple estimators in a single module, as long as they belong to the same 'family' of estimators.

In this case, I would argue that these Sensor classes should be included in a single module because they are strongly related to one another. From a user perspective it's also easier to use if you don't have to have a separate import for every single (related) sensor/operator/hook. 

The one-class-per-file preference is more java-esque and seems to be mainly preferred by (ex) Java programmers.